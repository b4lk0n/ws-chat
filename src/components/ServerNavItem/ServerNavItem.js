import React from 'react';
import { string, instanceOf, bool } from 'prop-types';
import { format } from 'date-fns';
import { Text } from '../Text';
import { ConnectionStatusIcon } from '../ConnectionStatusIcon';
import { Icon } from '../Icon';

import styles from './ServerNavItem.module.css';

export const ServerNavItem = ({ url, updatedAt, connectionStatus }) => {
  return (
    <div className={styles.root}>
      <Text>{url}</Text>
      {updatedAt ? <Text>{format(updatedAt, 'd LLL uuuu, HH:mm')}</Text> : null}

      <Icon value={<ConnectionStatusIcon success={connectionStatus} />} />
    </div>
  );
};
