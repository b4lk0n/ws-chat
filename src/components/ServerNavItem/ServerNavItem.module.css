.root {
  display: flex;
  align-items: center;
  padding: 14px 8px 14px 16px;
}

.clickable {
  cursor: pointer;
}

.info {
  flex: 1;
}

.serverUrl {
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 268px;
}

.updatedAt {
  color: var(--textSecondary);
}
