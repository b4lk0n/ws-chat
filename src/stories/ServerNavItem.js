import React from 'react';
import { withKnobs, text, select, date } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { ServerNavItem } from '../components';

export default {
  title: 'ServerNavItem',
  decorators: [withKnobs],
};

export const Default = () => (
  <div style={{ width: 320 }}>
    <ServerNavItem
      server={{
        url: text('url', 'wss://echo.websocket.org'),
        updatedAt: date('updatedAt', new Date()),
        connectionStatus: select('connectionStatus', {
          'â€”': undefined,
          success: true,
          failure: false,
        }),
      }}
      onClick={action('onClick')}
    />
  </div>
);
