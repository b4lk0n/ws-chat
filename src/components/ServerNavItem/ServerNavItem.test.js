import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ServerNavItem } from './ServerNavItem';

const timestamp = 1580469500994;
const server = {
  url: 'wss://echo.websocket.org',
  updatedAt: new Date(timestamp),
};

describe('<ServerNavItem />', () => {
  it('should renders correctly', () => {
    const { getByTestId } = render(<ServerNavItem server={server} />);
    expect(getByTestId('server-nav-item')).toMatchSnapshot();
  });

  it('should call onClick handler', () => {
    const handler = jest.fn();
    const { getByTestId } = render(<ServerNavItem server={server} onClick={handler} />);
    const item = getByTestId('server-nav-item');

    expect(handler).not.toBeCalled();
    fireEvent.click(item);
    expect(handler).toBeCalledTimes(1);
    expect(handler).toBeCalledWith(server);
  });
});
