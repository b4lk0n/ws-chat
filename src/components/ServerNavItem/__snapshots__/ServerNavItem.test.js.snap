// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerNavItem /> should renders correctly 1`] = `
<div
  class="root"
  data-testid="server-nav-item"
>
  <div
    class="info"
  >
    <span
      class="root undefined serverUrl"
      data-testid="text"
    >
      wss://echo.websocket.org
    </span>
    <span
      class="root caption updatedAt"
      data-testid="text"
    >
      31 Jan 2020, 13:18
    </span>
  </div>
  <div
    class="connectionStatus"
  >
    <span
      class="root undefined"
      data-testid="icon"
    >
      <span
        class="root"
        data-testid="connection-status-icon"
      />
    </span>
  </div>
</div>
`;
